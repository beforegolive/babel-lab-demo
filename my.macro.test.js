import pluginTester from 'babel-plugin-tester'
import plugin from 'babel-plugin-macros'

// it('测试自定义宏', () => {
pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: [
    `
      import MyMacro from './my.macro'
      const props = { arr:[{name:'123'}]}
      MyMacro(props, _=>_.arr[0].name)
    `,
  ],
})
// })
