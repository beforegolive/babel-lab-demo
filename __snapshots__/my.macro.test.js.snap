// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin 1. unknown plugin: 1. unknown plugin 1`] = `

import MyMacro from './my.macro'
const props = { arr:[{name:'123'}]}
// MyMacro(props, _=>_.arr[0].name)

      ↓ ↓ ↓ ↓ ↓ ↓

const props = {
  arr: [
    {
      name: "123",
    },
  ],
}; // MyMacro(props, _=>_.arr[0].name)


`;
